core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/join.kt

```diff
@@ -10,7 +10,9 @@ public fun <A, B> DataFrame<A>.join(
     other: DataFrame<B>,
     type: JoinType = JoinType.Inner,
     selector: JoinColumnsSelector<A, B>? = null
-): DataFrame<A> = joinImpl(other, type, true, selector)
+): DataFrame<A> {
+    return joinImpl(other, type, addNewColumns = type.addNewColumns, selector)
```

Comment:
probably provide name for each argument in the function, it looks odd like this :). Also, I'm fine with it being an expression function if it's just a one line thing

Reply:
wouldn't it look odd too? plus, it's not even me who wrote this initially :D 
```Kotlin
joinImpl(
  other = other, 
  type = type, 
  addNewColumns = type.addNewColumns,
  selector = selector
)
```

Reply:
no not at all XD I write functions always like that. IMO positional arguments are a nightmare for refactoring and I think originally they were only allowed áfter the positional ones so mixing them looks extra odd to me.